rpc/namenode.go

```diff
@@ -55,45 +67,96 @@ func (err *NamenodeError) Error() string {
 	return s
 }
 
-// NewNamenodeConnection creates a new connection to a Namenode, and preforms an
+type host struct {
+	address     string
+	lastFailure time.Time
+}
+
+// NewNamenodeConnection creates a new connection to a namenode and performs an
 // initial handshake.
 //
 // You probably want to use hdfs.New instead, which provides a higher-level
 // interface.
-func NewNamenodeConnection(address, user string) (*NamenodeConnection, error) {
-	conn, err := net.DialTimeout("tcp", address, connectTimeout)
-	if err != nil {
-		return nil, err
-	}
-
-	return WrapNamenodeConnection(conn, user)
+func NewNamenodeConnection(address string, user string) (*NamenodeConnection, error) {
+	return NewNamenodeConnectionWithOptions(NamenodeConnectionOptions{
+		Addresses: []string{address},
+		User:      user,
+	})
 }
 
-// WrapNamenodeConnection wraps an existing net.Conn to a Namenode, and preforms
-// an initial handshake.
-//
-// You probably want to use hdfs.New instead, which provides a higher-level
-// interface.
-func WrapNamenodeConnection(conn net.Conn, user string) (*NamenodeConnection, error) {
+// NewNamenodeConnectionWithOptions creates a new connection to a namenode with
+// the given options and performs an initial handshake.
+func NewNamenodeConnectionWithOptions(options NamenodeConnectionOptions) (*NamenodeConnection, error) {
```

Comment:
Unfortunately, because I'm dumb and made this exported way back when I started this library, we can't just remove the old function. Can you leave it and make it deprecated?